lib/ui/lerp.dart

```diff
@@ -6,29 +6,35 @@
 
 part of dart.ui;
 
-/// Linearly interpolate between two numbers.
-// TODO(cbracken): Consider making a and b non-nullable.
-//                 https://github.com/flutter/flutter/issues/64617
+/// Linearly interpolate between two numbers, `a` and `b`, by an extrapolation
+/// factor `t`.
+///
+/// When `a` and `b` are equal or both NaN, `a` is returned.  Otherwise, if
+/// `a`, `b`, and `t` are required to be finite or null, and the result of `a +
+/// (b - a) * t` is returned, where nulls are defaulted to 0.0.
 double? lerpDouble(num? a, num? b, double t) {
-  if (a == null && b == null)
-    return null;
+  if (a == b || (a?.isNaN ?? false) && (b?.isNaN ?? false))
+    return a as double?;
```

Comment:
won't this fail if a is an int? `a?.toDouble()` seems better. I guess we missed a test. :-)

Reply:
#20871 (where the above wasn introduced) hasn't landed yet, but sent #20880 which adds integer tests in the meantime.